storage/blockchain/src/config/config.rs

```diff
@@ -65,163 +57,137 @@ impl ConfigBuilder {
-        self.sync_mode = Some(sync_mode);
+    pub const fn reader_threads(mut self, reader_threads: ReaderThreads) -> Self {
+        self.reader_threads = Some(reader_threads);
         self
     }
 
-    /// Set a custom [`ReaderThreads`].
+    /// Tune the [`ConfigBuilder`] for the highest performing,
+    /// but also most resource-intensive & maybe risky settings.
+    ///
+    /// Good default for testing, and resource-available machines.
     #[must_use]
-    pub const fn reader_threads(mut self, reader_threads: ReaderThreads) -> Self {
-        self.reader_threads = Some(reader_threads);
+    pub fn fast(mut self) -> Self {
+        self.db_config =
+            cuprate_database::config::ConfigBuilder::new(Cow::Borrowed(cuprate_blockchain_dir()))
+                .fast();
+
+        self.reader_threads = Some(ReaderThreads::OnePerThread);
         self
     }
 
-    /// Set a custom [`ResizeAlgorithm`].
+    /// Tune the [`ConfigBuilder`] for the lowest performing,
+    /// but also least resource-intensive settings.
+    ///
+    /// Good default for resource-limited machines, e.g. a cheap VPS.
     #[must_use]
-    pub const fn resize_algorithm(mut self, resize_algorithm: ResizeAlgorithm) -> Self {
-        self.resize_algorithm = Some(resize_algorithm);
+    pub fn low_power(mut self) -> Self {
+        self.db_config =
+            cuprate_database::config::ConfigBuilder::new(Cow::Borrowed(cuprate_blockchain_dir()))
+                .low_power();
+
+        self.reader_threads = Some(ReaderThreads::One);
         self
     }
 }
 
 impl Default for ConfigBuilder {
     fn default() -> Self {
+        let db_directory = Cow::Borrowed(cuprate_blockchain_dir());
         Self {
-            db_directory: Some(Cow::Borrowed(cuprate_blockchain_dir())),
-            sync_mode: Some(SyncMode::default()),
+            db_directory: Some(db_directory.clone()),
+            db_config: cuprate_database::config::ConfigBuilder::new(db_directory),
             reader_threads: Some(ReaderThreads::default()),
-            resize_algorithm: Some(ResizeAlgorithm::default()),
         }
     }
 }
 
 //---------------------------------------------------------------------------------------------------- Config
-/// Database [`Env`](crate::Env) configuration.
+/// `cuprate_blockchain` configuration.
 ///
-/// This is the struct passed to [`Env::open`](crate::Env::open) that
-/// allows the database to be configured in various ways.
+/// This is a configuration built on-top of [`cuprate_database::config::Config`].
 ///
-/// For construction, either use [`ConfigBuilder`] or [`Config::default`].
+/// It contains configuration specific to this crate, plus the database config.
 ///
-// SOMEDAY: there's are many more options to add in the future.
+/// For construction, either use [`ConfigBuilder`] or [`Config::default`].
 #[derive(Debug, Clone, PartialEq, PartialOrd)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct Config {
```

Comment:
## `Config` 2/2
`cuprate_blockchain` also has a `Config`, which is built on-top of `cuprate_database`.

For now, the only extra data this contains is the `ReaderThreads` for the `tower::Service`.

Implementation detail: LMDB allows setting max reader threads so this is actually used in `cuprate_database` for that too (although it is passed as a `NonZeroUsize`). Not sure where the appropriate place is to define this type since it has a double meaning here. Maybe this should be documented...?